web/src/app/AccountApi.js

```diff
@@ -261,25 +261,25 @@ class AccountApi {
 
   async createBillingSubscription(tier, interval) {
     console.log(`[AccountApi] Creating billing subscription with ${tier} and interval ${interval}`);
-    return await this.upsertBillingSubscription("POST", tier, interval);
+    return this.upsertBillingSubscription("POST", tier, interval);
   }
 
   async updateBillingSubscription(tier, interval) {
     console.log(`[AccountApi] Updating billing subscription with ${tier} and interval ${interval}`);
-    return await this.upsertBillingSubscription("PUT", tier, interval);
+    return this.upsertBillingSubscription("PUT", tier, interval);
   }
```

Comment:
(`return await` is equivalent to async function `return` when there is no `catch`)

Reply:
I asked ChatGPT and I still don't fully understand, but this seems to be legit.

> In JavaScript, the await keyword is used to wait for a promise to resolve and return its resolved value. However, if a function or method is not using the await keyword, it means that it does not need to wait for the promise to resolve and can continue execution without blocking. Therefore, removing the await keyword in this case does not change the behavior of the code, and the change is considered equivalent.

Before I read this, I would have thought that "my function" returns the actual thing, whereas your function returns a promise. No?

Reply:
ChatGPT cont'd:

> The original version of the code uses the await keyword before this.upsertBillingSubscription("PUT", tier, interval), which indicates that the updateBillingSubscription method is an asynchronous function. It waits for the promise returned by upsertBillingSubscription to resolve before returning the resolved value.

> The modified version removes the await keyword, making it a regular function that returns a Promise. The Promise will resolve to the value returned by upsertBillingSubscription.

> So, the two versions are not completely equivalent in terms of the return value. The version with await returns the resolved value directly, while the version without await returns a Promise that resolves to the value.

Sooooo, your version returns the Promise, while mine returns the thing. But that doesn't matter because it has to be called with `await` anyway? Is that right?

Reply:
This is correct:

> The modified version removes the await keyword, making it a regular function that returns a Promise. The Promise will resolve to the value returned by upsertBillingSubscription.

This is not:

> So, the two versions are not completely equivalent in terms of the return value. The version with await returns the resolved value directly, while the version without await returns a Promise that resolves to the value.

The gist of is is that async/await is just syntactic sugar, it doesn't change how the function behaves when called.

The reason why `return await` is not required is that returning a promise and returning the awaited value of a promise results in exactly the same thing, unless you wrap it with try/catch and want to handle the error here.

<https://eslint.org/docs/latest/rules/no-return-await>

Reply:
I think I mostly get it. The link was really helpful too, thanks.